.github/workflows/rollback.yml

```diff
@@ -1,11 +1,287 @@
 name: Rollback
+run-name: Rollback the ${{ inputs.service }} service in the ${{ inputs.environment }} environment
 
 on:
   workflow_dispatch:
+    inputs:
+      service:
+        type: choice
+        description: The microservice which we want to rollback to the previous version.
+        options:
+          - inbound_mail
+          - api
+          - web
+          - webhook
+          - widget
+          - worker
+          - ws
+      environment:
+        type: choice
+        description: The environment where the service is located.
+        options:
+          - Development
+          - Production
+      region:
+        type: choice
+        description: The region of the environment. It's needed to know TF workspace. When you choose DEV environment this option is not needed.
+        options:
+          - [EU]
+          - [US]
+          - [EU,US]
+      mode:
+        type: choice
+        description: The Rollback mode. You can roll back to the previously deployed version or to the version that has the current commit hash of this branch in an image tag name or a deployment info.
+        options:
+          - Previous Version
+          - Commit Hash
 
 jobs:
   ecs:
+    if: contains(fromJson('["api", "inbound_mail", "webhook", "worker", "ws"]'), github.event.inputs.service)
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        region: ${{ fromJSON(github.event.inputs.region) }}
     timeout-minutes: 60
+    environment: ${{ github.event.inputs.environment }}
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
     steps:
-      - run: echo "Rolling back"
+      - run: echo "Rolling back ${{ github.event.inputs.service }} in ${{ github.event.inputs.environment }}"
+
+      - id: commit
+        if: contains(fromJson('["Commit Hash"]'), github.event.inputs.mode)
+        uses: prompt/actions-commit-hash@v3
+
+      - name: Prepare variables
+        id: variables
+        run: |
+          if [[ "${{ matrix.region }}" == "EU" && "${{ github.event.inputs.environment }}" == "Production" ]]; then
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏ I am curious, is there a way to declaring a static mapping object and avoid the if/elif logic in a Github action?

Reply:
> **question:** ‏ I am curious, is there a way to declaring a static mapping object and avoid the if/elif logic in a Github action?

It allows us to use matrix and not to write a lot of steps for each static value, I agree, it looks not super great but in this case it exclude extra part of code. Anyway, I got you @SokratisVidros  and I'll think on it.